api/python/t4/packages.py

```diff
@@ -218,9 +218,8 @@ def fetch(self, dest=None):
             None
         """
         if dest is None:
-            name = pathlib.Path(_to_singleton(self.physical_keys)).name
-            name = name.split('?versionId=')[0]
-            dest = name
+            pkey = _to_singleton(self.physical_keys)
+            dest = pathlib.PurePosixPath(unquote(urlparse(pkey).path)).name
 
         physical_key = _to_singleton(self.physical_keys)
```

Comment:
May want to just move this up and reuse it?

Reply:
Good eye, done.